src/shell.js

```diff
@@ -43,6 +43,20 @@ var Module = typeof {{{ EXPORT_NAME }}} !== 'undefined' ? {{{ EXPORT_NAME }}} :
 #include "promise_polyfill.js"
 #endif
 
+// See https://caniuse.com/mdn-javascript_builtins_object_assign
+#if MIN_CHROME_VERSION < 45 || MIN_EDGE_VERSION < 12 || MIN_FIREFOX_VERSION < 34 || MIN_IE_VERSION != TARGET_NOT_SUPPORTED || MIN_SAFARI_VERSION < 90000
```

Comment:
Just out of curiousity. I believe all of these version thresholds pre-date wasm support. So is this check just for supporting wasm2js/asmjs on old browsers? If that's the case, should the check be like 
```
if (WASM2JS && (MIN_VERSION_FOO < X || ....))
```

Reply:
Good point, if those versions also lack wasm support then this might only be relevant for wasm2js. We do something similar with `fetch` support (all browsers with wasm support fetch, so we assume it's always present if wasm is used). But we would need to check for WASM=2 (supporting both wasm2js and wasm) here, at least, and I'm not sure that simplifies things overall. Just having the caniuse.com version numbers here (as in the PR now) may be best as it's also easy to verify (at least assuming caniuse.com is correct).

Reply:
I just copied the numbers directly from caniuse.com.

I guess we could add the clause to the condition but its not clear to me why it would make it better?   Do you think it would help in some way?

Lets say I target and old version of firefox that has no WASM support... presumably we should fail the build unless WASM2JS is used.    If we don't do that already it still seems reasonable that output should conform to that version of firefix in other ways such as this.

Reply:
Yeah your point makes sense to me. Probably we'd just want a top level check somewhere, so that if you are targeting wasm without wasm2js, and targeting a min browser version that didn't actually support wasm, that the build fails. This may already exist

Reply:
Interesting - I don't think we have such a toplevel check, but I agree that it's a good idea!